app/View/Components/Community/UserCard.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\View\Components\Community;
+
+use App\Community\Enums\Rank;
+use App\Community\Enums\RankType;
+use App\Site\Enums\Permissions;
+use App\Site\Models\User as UserModel;
+use App\Support\Cache\CacheKey;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\View\Component;
+
+class UserCard extends Component
+{
+    private string|array $user;
+
+    public function __construct(string|array $user)
+    {
+        $this->user = $user;
+    }
+
+    public function render(): ?View
+    {
+        $username = $this->getUsername($this->user);
+
+        if (empty($username)) {
+            return null;
+        }
+
+        /* @var array $rawUserData */
+        $rawUserData = [];
+        if (is_array($this->user)) {
+            $rawUserData = $this->user;
+        } else {
+            $rawUserData = $this->getUserData($username);
+        }
+
+        if (!$rawUserData) {
+            return null;
+        }
+
+        $cardViewValues = $this->buildAllCardViewValues($username, $rawUserData);
+
+        return view('components.community.user-card', $cardViewValues);
+    }
+
+    private function getUsername(string|array $user): ?string
+    {
+        return is_string($user) ? $user : ($user['User'] ?? null);
+    }
+
+    private function getUserData(string $username): ?array
+    {
+        return Cache::store('array')->remember(
+            CacheKey::buildUserCardDataCacheKey($username . "1"),
```

Comment:
This seems prone to error. If appending "1" is meant to differentiate the behavior from some other cache entry, what happens if there is a user whose name is "$username1" and they read/write this key?

Reply:
Apologies, I was debugging something and forgot to back this out. "1" shouldn't be appended anymore as of https://github.com/RetroAchievements/RAWeb/pull/1674/commits/87b59e67ccb06ac73265d9fd3adbc75ac87d2ed4.